### PR TITLE
Implemented Sort/Argsort Ops in PyTorch

### PR DESCRIPTION
**Pull request by twaclaw**
## Description

* Implements `SortOp` in `PyTorch`
* Implements `ArgSortOp` in `PyTorch`
* :warning: passing `axis=None` throws an exception because `Reshape` is not yet implemented

## Related Issue

- [x] Related to #821 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by ricardoV94** [comment]
What happens with axis=None that introduces a reshape?

**Comment by twaclaw** [comment]
> What happens with axis=None that introduces a reshape?

I think it is this [line](https://github.com/pymc-devs/pytensor/blob/4ea96b272247d3dc6f82987f1433d59cea94af1c/pytensor/tensor/sort.py#L148). But I should rather remove that test and don't test for `axis=None` for the time being.

**Comment by ricardoV94** [comment]
I see, you can mark it as xfail with pytest.mark.xfail

File: tests/link/pytorch/test_sort.py [hunk 1]
```diff
@@ -7,17 +7,12 @@
 from tests.link.pytorch.test_basic import compare_pytorch_and_py
 
 
+@pytest.mark.xfail(reason="Reshape not implemented")
 @pytest.mark.parametrize("axis", [0, 1, None])
```
Comment (ricardoV94) [inline tests/link/pytorch/test_sort.py:11]: When only some of the parametrized conditions fail, you should use `pytest.param` to mark that specifically as xfailing. The others are working and shouldn't be marked. There are some examples in the tests somewhere

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**